localstorage/firebase-worker.js

```diff
@@ -23,21 +27,27 @@ var firebaseConfig = {
 
 /** The Firestore instance. */
 self.db = null;
-self.onmessage = async function(e) {
-  // Do Firebase work here.
+self.onmessage = function(e) {
   if (e.data === 'persistenceAvailable') {
-    await initializeApp(true);
+    initializeApp(/*withPersistence=*/true);
   } else if (e.data === 'persistenceUnavailable') {
-    await initializeApp(false);
+    initializeApp(/*withPersistence=*/false);
   } else if (e.data === 'enableNetwork') {
-    await self.db.enableNetwork().then(() => {
+    self.db.enableNetwork().then(() => {
       console.log('Enabled network');
     });
   } else if (e.data === 'disableNetwork') {
-    await self.db.disableNetwork().then(() => {
+    self.db.disableNetwork().then(() => {
       console.log('Disabled network');
     });
+  } else if (e.data === 'writeData') {
+    self.db.collection('followers').doc('user1').set({
+      name: 'Bobby',
+      // Set age to a random value to test that persistence works.
+      age: Math.floor(Math.random() * 80 + 20)
+    })
   }
+  // Add message handlers that do Firebase work here.
 };
 
 async function initializeApp(withPersistence) {
```

Comment:
You left this function as async even though the calling code doesn't await it or handle the promise result...